addon/-private/router-ext.js

```diff
@@ -0,0 +1,202 @@
+import Ember from 'ember';
+import emberRequire from './ext-require';
+
+const hasDefaultSerialize = emberRequire('ember-routing/system/route', 'hasDefaultSerialize');
+
+const {
+  Logger: {
+    info
+  },
+  Router,
+  RSVP,
+  assert,
+  get,
+  getOwner
+} = Ember;
+
+Router.reopen({
+  assetLoader: Ember.inject.service(),
+
+  /**
+   * We skip doing default query params stuff when attempting to go to an
+   * Engine route.
+   *
+   * @override
+   */
+  _prepareQueryParams(routeName) {
+    if (this._engineInfoByRoute[routeName]) {
+      return;
+    }
+
+    return this._super(...arguments);
+  },
+
+  /**
+   * We override this to fetch assets when crossing into an Engine. For other
+   * cases we do the normal thing.
+   *
+   * @override
+   */
+  _getHandlerFunction() {
+    let seen = Object.create(null);
```

Comment:
This is an approximation of what is happening upstream ([ref](https://github.com/emberjs/ember.js/blob/c94d277b82e57962953ccf601b8862016cb46f29/packages/ember-routing/lib/system/router.js#L563)). Will have it actually using `EmptyObject` in the next update.